### PR TITLE
heal context

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: docs/modules/plugins/partials/selenium-properties.adoc [hunk 1]
```diff
@@ -15,3 +15,8 @@ An empty property value will lead to the error: "Application session scope is no
 |true/false
 |true
 |Defines if Vividus publishes the application source code on failure
+
+|ui.context.self-healing
```
Comment (valfirst) [inline docs/modules/plugins/partials/selenium-properties.adoc:19]: also let's mark this feature as `incubating`

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

File: vividus-extension-selenium/src/main/java/org/vividus/ui/context/SelfHealingUiContext.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    private <T extends SearchContext> T wrap(T searchContext, Class<?> clazz, Supplier<T> searchContextGetter)
+    {
+        if (!(searchContext instanceof WebElement) || !clazz.isInterface())
+        {
+            return searchContext;
+        }
+        Set<Class<?>> proxied = new HashSet<>();
+        proxied.add(WrapsElement.class);
+        proxied.add(WebElement.class);
+        proxied.add(clazz);
```
Comment (uarlouski) [inline vividus-extension-selenium/src/main/java/org/vividus/ui/context/SelfHealingUiContext.java:59]: Class<?>[] proxied = { WrapsElement.class, WebElement.class, clazz };
mb?

Comment (ikalinin1) [reply vividus-extension-selenium/src/main/java/org/vividus/ui/context/SelfHealingUiContext.java:59]: It is a set for a reason :-)

Comment (ikalinin1) [reply vividus-extension-selenium/src/main/java/org/vividus/ui/context/SelfHealingUiContext.java:59]: It is a set for a reason :-)

Comment (uarlouski) [reply vividus-extension-selenium/src/main/java/org/vividus/ui/context/SelfHealingUiContext.java:59]: which reason? :)

Comment (ikalinin1) [reply vividus-extension-selenium/src/main/java/org/vividus/ui/context/SelfHealingUiContext.java:59]: `searchContext.getSearchContext(WebElement.class)`

Comment (ikalinin1) [reply vividus-extension-selenium/src/main/java/org/vividus/ui/context/SelfHealingUiContext.java:59]:
```
repeated interface: org.openqa.selenium.WebElement
java.lang.IllegalArgumentException: repeated interface: org.openqa.selenium.WebElement
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.validateProxyInterfaces(Proxy.java:720)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.<init>(Proxy.java:646)
	at java.base/java.lang.reflect.Proxy.lambda$getProxyConstructor$1(Proxy.java:439)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue$Memoizer.get(AbstractClassLoaderValue.java:329)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue.computeIfAbsent(AbstractClassLoaderValue.java:205)
	at java.base/java.lang.reflect.Proxy.getProxyConstructor(Proxy.java:437)
	at java.base/java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1028)
```

**Review by uarlouski — APPROVED** [review]

**Pull request merged**